src/main/scala/org/broadinstitute/dsde/agora/server/business/AgoraBusiness.scala

```diff
@@ -370,15 +370,30 @@ class AgoraBusiness(permissionsDataSource: PermissionsDataSource)(implicit ec: E
           parseWDL(wdl) match {
             case Failure(ex) => throw ex
             case Success(workflow) =>
-              // get the set of input and output keys for this WDL
-              val inputKeys:Set[String] = workflow.inputs.keySet.map(_.toString)
+              // get the set of required input keys, optional input keys, and output keys for this WDL
+              val (optionalInputs, requiredInputs) = workflow.inputs.partition(_._2.optional)
+              val optionalInputKeys = optionalInputs.keySet.map(_.toString)
+              val requiredInputKeys = requiredInputs.keySet.map(_.toString)
               val outputKeys:Set[String] = workflow.outputs.map(_.fullyQualifiedName.toString).toSet
 
-              // filter configs to those that have the exact same input and output keys
+              // define "compatible" to be:
+              //  - config has exact same set of output keys as method's WDL
+              //  - config has all required input keys in the method's WDL
+              //  - any additional input keys in the config must exist in the set of optional WDL inputs
               configs.filter { config =>
                 config.payloadObject match {
                   case None => false
-                  case Some(mc) => mc.outputs.keySet == outputKeys && mc.inputs.keySet == inputKeys
+                  case Some(mc) =>
+                    // do outputs match?
+                    val outputsMatch = mc.outputs.keySet == outputKeys
+                    // are all required inputs satisfied?
+                    val requiredsMatch = (requiredInputKeys diff mc.inputs.keySet).isEmpty
```

Comment:
If all entries in `requiredInputKeys` are present in `mc.inputs.keySet`, but there are extra (optional) entries in `mc.inputs.keySet`, `requiredsMatch` will always be `false`. Is this right?